internal/auth0/resourceserver/resource_scope.go

```diff
@@ -0,0 +1,219 @@
+		Description: "With this resource, you can manage scopes (permissions) associated with a resource server (API).",
+	}
+}
+
+func createResourceServerScope(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	resourceServerID := data.Get("resource_server_identifier").(string)
+	scope := data.Get("scope").(string)
+	description := data.Get("description").(string)
+
+	currentScopes, err := api.ResourceServer.Read(resourceServerID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	scopes := append(currentScopes.GetScopes(), management.ResourceServerScope{
+		Value:       &scope,
+		Description: &description,
+	})
+	resourceServer := management.ResourceServer{
+		Scopes: &scopes,
+	}
+
+	if err := api.ResourceServer.Update(resourceServerID, &resourceServer); err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(fmt.Sprintf(`%s::%s`, resourceServerID, scope))
+
+	return readResourceServerScope(ctx, data, meta)
+}
+
+func updateResourceServerScope(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	resourceServerID := data.Get("resource_server_identifier").(string)
+	scope := data.Get("scope").(string)
+	newDescription := data.Get("description").(string)
+
+	existingScopes, err := api.ResourceServer.Read(resourceServerID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	updatedScopes := []management.ResourceServerScope{}
+
+	found := false
+	for _, p := range existingScopes.GetScopes() {
+		updated := p
+		if p.GetValue() == scope {
+			found = true
+			updated.Description = &newDescription
+		}
+		updatedScopes = append(updatedScopes, updated)
+	}
+
+	if !found {
+		data.SetId("")
+		return nil
+	}
+
+	if err := api.ResourceServer.Update(resourceServerID, &management.ResourceServer{
+		Scopes: &updatedScopes,
+	}); err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(fmt.Sprintf(`%s::%s`, resourceServerID, scope))
+
+	return readResourceServerScope(ctx, data, meta)
+}
+
+func readResourceServerScope(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	resourceServerID := data.Get("resource_server_identifier").(string)
+	scope := data.Get("scope").(string)
+
+	existingScopes, err := api.ResourceServer.Read(resourceServerID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	for _, p := range existingScopes.GetScopes() {
+		if p.GetValue() == scope {
+			result := multierror.Append(
+				data.Set("description", p.GetDescription()),
```

Comment:
We're missing setting the resource server identifier and scope properties, otherwise this resource will have the same faulty import behavior as described in https://github.com/auth0/terraform-provider-auth0/issues/593, so let's also add a test case similar to https://github.com/auth0/terraform-provider-auth0/pull/594

Reply:
Only resources that have ```		Importer: &schema.ResourceImporter{
			StateContext: schema.ImportStatePassthroughContext,
		},``` have this faulty behavior. I'll patch those in a separate PR.

Reply:
Update: We're gonna focus on refactoring the import tests and add more in a separate PR.